run_molecule_tests.py

```diff
@@ -0,0 +1,78 @@
+import logging
+import os
+import subprocess
+from collections.abc import Sequence
+from pathlib import Path
+from typing import Set
+
+
+def run_command(
+    cmd: Sequence[str], *, stdout: int | None = subprocess.PIPE, cwd: Path | None = None
+) -> subprocess.CompletedProcess[str]:
+    logging.info("running command: %s", cmd)
+    return subprocess.run(cmd, stdout=stdout, check=True, text=True, cwd=cwd)
+
+
+def get_diff(origin: str, current: str) -> Sequence[Path]:
+    """Return paths of changed files between two git branch / hash, that list
+    is simply based on git diff result"""
+    return [
+        Path(f)
+        for f in (
+            run_command(
+                ["git", "diff-tree", "--name-only", "-r", origin, current]
+            ).stdout.splitlines()
+        )
+    ]
+
+
+def get_subdirs(
+    parents: Sequence[Path], potential_children: Sequence[Path]
+) -> Set[str]:
+    """Return names set direct sub directory (one level under any parents) for
+    which there is at least one entry on potential_children.
+
+    >>> parent_locations = [Path('r/f/')]
+    >>> potential_children = [Path('r/f/good'), Path('t/'), Path('r/b/')]
+    >>> get_subdirs(parent_locations, potential_children)
+    {'good'}
```

Comment:
```suggestion
    >>> parent_locations = [Path('r/f/')]
    >>> potential_children = [Path('r/f/good/foo'), Path('t/'), Path('r/b/')]
    >>> get_subdirs(parent_locations, potential_children)
    {'good'}
```

That would show that it's only the next hierarchy level that we want.